app/main.py

```diff
@@ -2,5 +1,14 @@
+from typing import Callable, Any
 
 
-def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+def cache(func: callable) -> Callable:
+    cache_result = {}
+
+    def function_cache(*args) -> Any:
+        if args not in cache_result:
+            cache_result[*args] = func(*args)
+            print("Calculating new result")
+        else:
+            print("Getting from cache")
+        return cache_result[*args]
```

Comment:
Similar to the previous comment, use `cache_result[args]` instead of `cache_result[*args]` to access the cached result.